src/apps/backend/modules/task/rest-api/task-router.ts

```diff
@@ -12,7 +12,8 @@ export default class TaskRouter extends ApplicationRouter {
 
     router.post('/', ctrl.createTask);
     router.get('/', ctrl.getTasks);
-    router.get('/:id', ctrl.getTask);
+    router.get('/shared/:id', ctrl.getSharedTasks);
```

Comment:
Why are we creating a separate API to get shared tasks? Just add a query param `sharedTasks` to the API that is used to get all tasks and return all shared tasks to that API only.